rustler/src/types/atom.rs

```diff
@@ -137,6 +138,15 @@ pub(in crate::types) fn decode_bool(term: Term) -> NifResult<bool> {
     Err(Error::BadArg)
 }
 
+impl Hash for Atom {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        use crate::sys::{enif_hash, ErlNifHash};
+        let hash =
+            unsafe { enif_hash(ErlNifHash::ERL_NIF_INTERNAL_HASH, self.as_c_arg(), 0) as u32 };
```

Comment:
Question: This somewhat duplicates the code in Term, should we typecast somehow to reuse the same code? Or do we need to keep the code separate?